rpcs3/Loader/mself.hpp

```diff
@@ -12,10 +29,10 @@ struct mself_header
 	be_t<u32> header_size; // ???
 	u8 reserved[0x28];
 
-	u32 get_count(u64 file_size)
+	u32 get_count(u64 file_size) const
 	{
 		// Fast sanity check
-		if (magic != "MSF"_u32 || ver != u32{1} || this->size != file_size) [[unlikely]]
+		if (magic != "MSF"_u32 || ver != u32{1} || (file_size - sizeof(mself_header)) / sizeof(mself_record) < count || this->size != file_size) [[unlikely]]
```

Comment:
does this handle overflow if file_size < 0x40 ?

Reply:
It must be over 0x40 because otherwise the file header could not be read.